### PR TITLE
5230 haru

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

(Details)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/model.py [hunk 1]
```diff
@@ -22,12 +23,42 @@
...
+            "model_uuid",
+            "saved_input_example_info",
+            "signature",
+            "input_schema",
+            "output_schema",
+        ],
+    )
+):
+    """
+    I am ModelInfo
```
Comment (dbczumar) [inline mlflow/models/model.py:50]: <3

File: docs/source/python_api/mlflow.models.rst [hunk 2]
```diff
@@ -5,4 +5,7 @@ mlflow.models
     :members:
     :undoc-members:
     :show-inheritance:
+    :exclude-members: ModelInfo
 
+.. autoclass:: mlflow.models.ModelInfo
+    :members:
```
Comment (harupy) [inline docs/source/python_api/mlflow.models.rst:11]: I haven't tried this but this should also work.

```suggestion
```

File: mlflow/models/__init__.py [hunk 3]
```diff
@@ -21,13 +21,14 @@
 For details, see `MLflow Models <../models.html>`_.
 """
 
-from .model import Model
+from .model import Model, ModelInfo
```
Comment (harupy) [inline mlflow/models/__init__.py:24]: Import `ModelInfo` here.

File: mlflow/models/model.py [hunk 4]
```diff
@@ -109,6 +140,24 @@ def saved_input_example_info(self, value: Dict[str, Any]):
         # pylint: disable=attribute-defined-outside-init
         self._saved_input_example_info = value
 
+    def get_model_info(self):
+        """
+        Create a :py:class:`ModelInfo <mlflow.models.ModelInfo>` instance that contains the model
+        metadata.
+        """
```
Comment (harupy) [inline mlflow/models/model.py:147]: Added a reference to `mlflow.models.ModelInfo` here.

File: docs/source/python_api/mlflow.models.rst (hunk 2 above)
Comment (harupy) [reply docs/source/python_api/mlflow.models.rst:11]: I found `show-inheritance` causes sphinx to complain:

```
/Users/harutakakawamura/Desktop/repositories/mlflow/mlflow/models/model.py:docstring of mlflow.models.model.ModelInfo:1: WARNING: py:class reference target not found: mlflow.models.model.ModelInfo
```

**Pull request closed without merging**